include/deal.II/base/function_spherical.h

```diff
@@ -0,0 +1,122 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__function_spherical_h
+#define dealii__function_spherical_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/std_cxx11/array.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Functions
+{
+
+  /**
+   * An abstract base class for a scalar-valued function $f=f(r,\theta,\phi)$
+   * defined in spherical coordinates. This class wraps transformation of values,
+   * gradients and hessians from spherical coordinates to Cartesian coordinate system.
+   * Therefore derived classes only need to implement those functions in
+   * spherical coordinates (specifically svalue(), sgradient() and
+   * shessian() ). The convention for angles is the same as in
+   * GeometricUtilities::Coordinates.
+   *
+   * @note This function is currently only implemented for dim==3 .
+   *
+   * @author Denis Davydov, 2017
+   */
+  template <int dim>
+  class Spherical : public Function<dim>
+  {
+  public:
+    /**
+     * Constructor which should be provided with @p center defining the origin
+     * of the coordinate system.
+     */
+    Spherical(const Point<dim> &center);
+
+    /**
+     * Virtual destructor.
+     */
+    virtual ~Spherical();
+
+    /**
+     * Return the value of the function at the given point.
+     */
+    virtual double value (const Point<dim> &point,
+                          const unsigned int component = 0) const;
+
+    /**
+     * Return the gradient in Cartesian coordinates at point @p p.
+     */
+    virtual Tensor<1,dim> gradient (const Point<dim>   &p,
+                                    const unsigned int  component = 0) const;
+
+    /**
+     * Return the Hessian in Cartesian coordinates at point @p p.
+     */
+    virtual SymmetricTensor<2,dim> hessian (const Point<dim> &p,
+                                            const unsigned int component=0) const;
+
+    std::size_t memory_consumption () const;
+
+  private:
+    /**
+     * Return the value at point @p sp.
+     */
+    virtual double svalue(const std_cxx11::array<double, dim> &sp) const = 0;
```

Comment:
say something like "Here, @p sp is provided in spherical coordinates."

By the way, why not add the `component` argument the `value()` function has?